src/pint/models/jump.py

```diff
@@ -172,12 +178,12 @@ def jump_params_to_flags(self, toas):
             for d in toas.table["flags"][mask]:
                 if "jump" in d:
                     index_list = d["jump"].split(",")
-                    if str(jump_par.index) in index_list:
```

Comment:
What was this part doing that was problematic?

Reply:
Or is it just making sure that all TOAs that are affected by all JUMPs always have their flags re-written?  If so, I'm just wondering if the previous behavior caused an issue?

Reply:
I had to rewrite this because a test was failing when I rewrote `jump_flags_to_params`. Essentially, this never removed a jump's number from the list - anythjng that was ever in the `-jump` flag always stayed there unless some other function took it out. Now this function can be (should be!) the only thing that ever modifies the `-jump` flag. 

(I still maintain that any function that wants to look at the `-jump` flag is better off just running essentially this function to find out what JUMPs they need to worry about. But since the `-jump` flag is only used in pintk I'm not yet confident I can make that change. It's still probably a good idea for any function that wants to look at the `-jump` flag to run this first to make sure it's up-to-date and no one has added a JUMP or a TOA or changed anything.)

The `if` statement here is harmless but since I'm starting from nothing and iterating over all the JUMPs, it's never true.